### PR TITLE
Fix clang-tidy findings

### PR DESCRIPTION
**Pull request by masterleinad**
Related to #17202. I didn't apply any changes regarding [performance-enum-size](https://clang.llvm.org/extra/clang-tidy/checks/performance/enum-size.html). Otherwise, this addresses all clang-tidy findings with clang++-18.1.5.

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

File: source/base/parameter_acceptor.cc [hunk 1]
```diff
@@ -59,12 +59,7 @@ ParameterAcceptor::ParameterAcceptor(const std::string &name)
 ParameterAcceptor::~ParameterAcceptor()
 {
   std::lock_guard<std::mutex> l(class_list_mutex);
-  // Notice that it is possible that the class is no longer in the static list.
-  // This happens when the clear() method has been called. We must guard that
-  // this is not the case, and therefore we only remove ourselves from the list
-  // if we are actually there.
-  if (class_list.find(this) != class_list.end())
-    class_list.erase(this);
+  class_list.erase(this);
```
Comment (tjhei) [inline source/base/parameter_acceptor.cc:62]: It might make sense to leave a comment explaining that we remove ourselves if we are still registered here.

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Shall we keep enum size warnings enablednif we don't fix them? I don't know but there might be some place where this actually matters to reduce the size of large objects (Tria?).

File: source/base/parameter_acceptor.cc (hunk 1 above)
Comment (masterleinad) [reply source/base/parameter_acceptor.cc]: This should have (only) been a part of #17208.

**Comment by masterleinad** [comment]
> Shall we keep enum size warnings enablednif we don't fix them? I don't know but there might be some place where this actually matters to reduce the size of large objects (Tria?).

The check was only introduced in 18.1.0, so we don't see it in the CI. I think we can keep it in and decide later.

**Pull request merged**